### PR TITLE
Codespeak refactor, serb codes [TEST]

### PR DESCRIPTION
**Pull request by Humonitarian**
## About The Pull Request

Refactors codespeak to be decodable, adds serb codes (codes are randomized inbetween rounds)

## Why It's Good For The Game

More code shenanigans

## Changelog
:cl:
add: Serbian codewords and codespeak perk
tweak: Codespeak can be decoded (randomized at roundstart)
/:cl:

### CODE REVIEW

File: code/datums/perks/job.dm [hunk 1]
```diff
@@ -294,6 +294,7 @@
 		/mob/living/carbon/human/proc/codespeak_warcrime_yes_local,
 		/mob/living/carbon/human/proc/codespeak_warcrime_no_local,
 		/mob/living/carbon/human/proc/codespeak_run_local)
+	var/codetype = "IH"
```
Comment (SirRichardFrancis) [inline code/datums/perks/job.dm:297]: Is it used somewhere?

File: code/game/jobs/job/security.dm [hunk 2]
```diff
@@ -29,8 +29,8 @@
 		STAT_VIG = 40,
 	)
 
-	perks = list(/datum/perk/survivor,
-				 /datum/perk/codespeak)
+	perks = list(/PERK_SURVIVOR,
```
Comment (SirRichardFrancis) [inline code/game/jobs/job/security.dm:32]:
```suggestion
	perks = list(PERK_SURVIVOR,
```

**Review by SirRichardFrancis — APPROVED** [review]
Need more variations of "Idiot" code, it's gonna be most popular.

**Comment by ErisMeleeEnjoyer** [comment]
tfw idiot is local only

File: code/datums/perks/job.dm (hunk 1 above)
Comment (Humonitarian) [reply code/datums/perks/job.dm:297]: I don't actually think so

Comment (SirRichardFrancis) [reply code/datums/perks/job.dm:297]: Then maybe remove it idk.

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]
Wait, is Serbian codespeak identical to Ironhammer codespeak?
Oh boooy, you're missing on a huge meme potential here.

**Comment by Alliostra** [comment]
> Wait, is Serbian codespeak identical to Ironhammer codespeak? Oh boooy, you're missing on a huge meme potential here.

Meme? no fun allowed in codespeak. sorry.

**Comment by SirRichardFrancis** [comment]
> Meme? no fun allowed in codespeak. sorry.

:clueless:

**Comment by Humonitarian** [comment]
> Wait, is Serbian codespeak identical to Ironhammer codespeak? Oh boooy, you're missing on a huge meme potential here.

Not missing. The codes themselves can randomly generate to be the same. There is a definite chance that when serbs say one of the codes, IH will understand it as if it were their own.

**Comment by SirRichardFrancis** [comment]
> Not missing. The codes themselves can randomly generate to be the same. There is a definite chance that when serbs say one of the codes, IH will understand it as if it were their own.

Missing. At least change the structure of the codes for Serbs. Two unrelated organizations with no lore-backed affiliation, history or even any proximity to speak of, using exactly the same code language? Rly? That's very, _very_ lazy.

**Comment by driecg36** [comment]
> Missing. At least change the structure of the codes for Serbs. Two unrelated organizations with no lore-backed affiliation, history or even any proximity to speak of, using exactly the same code language? Rly? That's very, _very_ lazy.

10 codes are a widespread practice used all over the world today. Militaries intentionally use the same shorthands and terminology in order to communicate more efficiently. How is it unrealistic at all that two paramilitary organizations would end up using similar variations of a widespread code language used in that field?

**Comment by SirRichardFrancis** [comment]
> 10 codes are a widespread practice used all over the world today. Militaries intentionally use the same shorthands and terminology in order to communicate more efficiently. How is it unrealistic at all that two paramilitary organizations would end up using similar variations of a widespread code language used in that field?

_**Allied**_ militaries intentionally use the same shorthands and terminology in order to communicate more efficiently.
Different military alliances known to come up with "reporting names"/nicknames for shit that already has a name.
Difference in stuff like code language for secure communication shouldn't be even mentioned, due to how obvious it is, but here we are...

Anyway, while coming up with codespeak format I needed something associated with America, to comply with faction theme.
So NATO phonetic alphabet and APCO brevity codes (aka police ten-codes) were chosen.
Maybe it's just me, but I only know of USA cops using codes like that, and maybe people who use radio in 2022.

**Comment by maikilangiolo** [comment]
Ok but like, a drunk IHC sent a broadcast by mistake and Serbs picked it up

**Comment by SirRichardFrancis** [comment]
Ok and?

**Comment by maikilangiolo** [comment]
It's why they share the codespeak, but only understanding some words.

**Comment by SirRichardFrancis** [comment]
That's seriously your entire point?

File: code/modules/mob/language/codespeak.dm [hunk 3]
```diff
@@ -1,42 +1,86 @@
...
+	if(codetype == "IH")
+		var/cop_code_1 = pick("10", "20", "0", "13")
+		var/cop_code_2 = pick("1","4", "7", "8", "10", "13", "17", "21", "22", "24", "33", "42", "54", "64", "75", "88", "99")
+		var/cop_code_3 = pick("Alpha", "Bravo", "Charlie", "Delta", "Echo", "Foxtrot", "Sierra", "Uniform")
+		return "[cop_code_1]-[cop_code_2] [cop_code_3]"
+	else if(codetype == "SA")
+		var/cop_code_3 = pick("Anna", "Boris", "Vasily", "Grigory", "Dmitry", "Yelena", "Zhenya", "Ivan")
+		var/cop_code_1 = "[pick("1", "2", "3", "5", "7")]-"
+		var/cop_code_2 = pick("Konstatin", "Leonid", "Mikhail", "Nikolai", "Olga", "Pavel")
+		return "[cop_code_1]-[cop_code_2] [cop_code_3]"
```
Comment (SirRichardFrancis) [inline code/modules/mob/language/codespeak.dm:15]:
```suggestion
		var/cop_code_1 = pick("Alil-Aga", "Boris", "Cvijan", "Dimitrije", "Grigorije", "Jelena", "Vasilij", "Leonid", "Nikolaj")
		var/cop_code_2 = pick("Srbije", "Rakija", "Cevapi", "Tito", "Artiljerija", "Budala", "Slava")
		return "[cop_code_1]! [cop_code_2]!"
```

**Review by SirRichardFrancis — APPROVED** [review]

**Comment by SirRichardFrancis** [comment]
> Not approving until all code words are serbian memes

If only I knew serbian memes...

**Comment by Pink-Chink** [comment]
I guess we'll live test this.

**Review by Pink-Chink — APPROVED** [review]

**Pull request merged**